### DISCUSSION TITLE
Possibly incorrect react experimental rollback

### DISCUSSION THREAD

**Discussion by AdrienLemaire** [renovatebot/renovate#8095] [Request Help]
**What Renovate type, platform and version are you using?**

**Describe the bug**
Got 2 renovate PRs to rollback react & react-dom

| Package | Type | Update | Change |
|---|---|---|---|
| [react](https://reactjs.org/) ([source](https://togithub.com/facebook/react)) | dependencies | rollback | [`0.0.0-experimental-4ead6b530` -> `0.0.0-experimental-3310209d0`](https://renovatebot.com/diffs/npm/react//) |
| [react-dom](https://reactjs.org/) ([source](https://togithub.com/facebook/react)) | dependencies | rollback | [`0.0.0-experimental-4ead6b530` -> `0.0.0-experimental-3310209d0`](https://renovatebot.com/diffs/npm/react-dom//) |

Reverting broke my tests with numerous errors:

> Minified React error #170; visit https://reactjs.org/docs/error-decoder.html?invariant=170 for the
> full message or use the non-minified dev environment for full errors and additional helpful warnings.

Checking the react github repo and commit logs, I cannot find any mention of a revert needed, and I cannot even find the commit which hash starts with 331020.
Why is this rollback done, and what is is rollbacking to ?

**To Reproduce**

setup a package.json with 
```json
    "react": "0.0.0-experimental-4ead6b530",
    "react-dom": "0.0.0-experimental-4ead6b530",
```
renovate has then created the PR to rollback to 3310209d0

**Comment by rarkins** [comment]
The way these experimental versions are done is a misuse of semver, and they will be sorted as per semver guidelines. It's hard to tell what is the correct one to be using because you can't sort commit hashes, but here's the current tagging:

![image](https://user-images.githubusercontent.com/6311784/102759920-13f52680-4375-11eb-8002-9cc550f383ab.png)

Here's the recent list of published versions:

![image](https://user-images.githubusercontent.com/6311784/102759962-23746f80-4375-11eb-9997-a54a1d3d637a.png)

I suggest you utilize `followTag` and set it to "experimental" for the react page (use a package rule to configure it).

**Reply by AdrienLemaire** [reply]
@rarkins thanks for the reply!

I actually already have the followTag. 

```json
[
    {
      "followTag": "experimental",
      "ignoreUnstable": false,
      "packageNames": [
        "react",
        "react-dom",
        "react-router",
        "react-router-dom"
      ]
    }
  ]
```

If I understand correctly, the PR message was misleading because it's not a rollback but an upgrade. And I need to fix my repo.
Indeed, I can verify that at https://www.npmjs.com/package/react.
Looks like the react team's use of git tag is not reliable.

Thanks!

**Reply by rarkins** [reply]
I guess what happened is:
- Renovate has correctly detected/followed the right "version"
- It calculates that the new version is *less* than the current version so classifies it as a rollback.
